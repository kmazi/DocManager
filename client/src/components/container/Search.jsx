import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';

import { searchDocuments } from '../../actions/documentActions';
import Search from '../presentation/Search';

export default
connect(null, { searchDocuments })(withRouter(Search));
