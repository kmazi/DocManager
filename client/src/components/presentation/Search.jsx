import React from 'react';
import PropTypes from 'prop-types';
import $ from 'jquery';

const Search = ({ searchDocuments }) => (
  <div className="searchcontainer">
    <div >
      <input
        type="text"
        onChange={() => {
          const searchText = $('.searchcontainer input').val();
          searchDocuments(searchText);
        }}
        placeholder="Search for documents"
      />
    </div>
  </div>
);

Search.propTypes = {
  searchDocuments: PropTypes.func.isRequired,
};

export default Search;
