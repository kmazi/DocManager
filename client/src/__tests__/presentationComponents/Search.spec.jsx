import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import Search from '../../components/presentation/Search';

describe('The search box component:', () => {
  const props = {
    searchDocuments: jest.fn(() => Promise.resolve()),
  };
  const enzymeWrapper = mount(<Search {...props} />);
  test('should display bold white bottom border when focused', () => {
    const searchDocuments = jest.fn(() => Promise.resolve());
    const component = renderer.create(
      <Search searchDocuments={searchDocuments} />
  );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should submit the search text when input changes', () => {
    enzymeWrapper.find('.searchcontainer input').props().onChange();
    expect(props.searchDocuments.mock.calls.length).toBe(1);
  });
});
