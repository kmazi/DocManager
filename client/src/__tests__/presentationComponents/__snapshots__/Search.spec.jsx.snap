// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The search box component: should display bold white bottom border when focused 1`] = `
<div
  className="searchcontainer"
>
  <div>
    <input
      onChange={[Function]}
      placeholder="search all documents"
      type="text"
    />
  </div>
</div>
`;
